src/application/service/AddressLookUpService.ts

```diff
@@ -33,43 +34,90 @@ class AddressLookUpService {
         postalCode
       );
 
-      if (isValid && addressLine1) {
-        const ukAddresses: UKAddress[] =
-          await this.addressGateway.getListOfValidPostcodeAddresses(
-            opt,
-            postalCode
-          );
+      if (!isValid) {
+        this.setPostalCodeError(
+          uiErrors,
+          `The postcode ${postalCode} cannot be found`
+        );
+
+        return { address, errors: uiErrors };
+      }
+
+      const ukAddresses: UKAddress[] =
+        await this.addressGateway.getListOfValidPostcodeAddresses(
+          opt,
+          postalCode
+        );
 
+      if (!this.isFromCorrectCountry(partnershipType, ukAddresses, uiErrors)) {
+        return { address, errors: uiErrors };
+      }
+
+      if (premise) {
         if (ukAddresses.length === 0) {
-          return {
-            isValid,
-            address
-          };
+          return { address };
         }
 
         const matchingAddress = ukAddresses.find(
           (ukAddress) =>
-            ukAddress.postcode === postalCode &&
-            ukAddress.addressLine1 === addressLine1
+            ukAddress.postcode === postalCode && ukAddress.premise === premise
         );
 
         if (matchingAddress) {
-          return { isValid, address: matchingAddress };
+          return { address: matchingAddress };
         }
       }
 
-      return {
-        isValid,
-        address
-      };
+      return { address };
     } catch (error: any) {
       logger.error(`Error validating postcode ${JSON.stringify(error)}`);
 
       throw error;
     }
   }
 
-  private;
+  private isFromCorrectCountry(
+    partnershipType: string,
+    ukAddresses: UKAddress[],
+    uiErrors: UIErrors
+  ): boolean {
+    let isCorrectCountry = true;
+
+    const SCOTLAND = ["GB-SCT", "DG", "NE", "CA", "TD"];
+    const IS_IN_SCOTLAND = SCOTLAND.includes(ukAddresses[0]?.country);
+    const SCOTLAND_TYPE =
+      partnershipType === PartnershipType.SLP ||
+      partnershipType === PartnershipType.SPFLP;
+    const NON_SCOTLAND_TYPE =
+      partnershipType === PartnershipType.LP ||
+      partnershipType === PartnershipType.PFLP;
+
+    if (SCOTLAND_TYPE && !IS_IN_SCOTLAND) {
+      isCorrectCountry = false;
+
+      this.setPostalCodeError(
+        uiErrors,
+        "You must enter a postcode which is in Scotland"
+      );
+    } else if (NON_SCOTLAND_TYPE && IS_IN_SCOTLAND) {
+      isCorrectCountry = false;
+
+      this.setPostalCodeError(
+        uiErrors,
+        "You must enter a postcode which is in England, Wales, or Northern Ireland"
```

Comment:
Would it be better to have the error messages defined in a separate file, that way we can reference them in the unit tests as well.

Reply:
thinking about it, perhaps it would be good to have the unit test use it's own string, as there could be a mistake in the imported error message and the test would pass.
leave it as it is.